### PR TITLE
Streaming media embed

### PR DESCRIPTION
**Pull request by Rijidij**

### CODE REVIEW

File: etna/media/blocks.py [hunk 1]
```diff
@@ -3,34 +3,23 @@
...
-from wagtailmedia.blocks import ChooserBlock, AbstractMediaChooserBlock
+from wagtail.images.blocks import ImageChooserBlock
+from wagtailmedia.blocks import AbstractMediaChooserBlock
 
 
 class EtnaMediaBlock(AbstractMediaChooserBlock):
     def render_basic(self, value, context=None):
-        if not value:
-            return ""
+        pass
```
Comment (danbentley) [inline etna/media/blocks.py:12]: Would you mind adding a comment about why this method is required even though it does nothing.

I checked the wagtail source, render_basic is only called if no template is defined:

https://github.com/wagtail/wagtail/blob/8413d00bdd03c447900019961d604186e17d2870/wagtail/core/blocks/base.py#L206

File: etna/media/blocks.py [hunk 2]
```diff
@@ -3,34 +3,23 @@
 from django.template.loader import render_to_string
 
 from wagtail.core import blocks
-from wagtailmedia.blocks import ChooserBlock, AbstractMediaChooserBlock
+from wagtail.images.blocks import ImageChooserBlock
+from wagtailmedia.blocks import AbstractMediaChooserBlock
 
 
 class EtnaMediaBlock(AbstractMediaChooserBlock):
```
Comment (danbentley) [inline etna/media/blocks.py:10]: Would you mind renaming to `MediaChooserBlock`?

File: etna/media/templatetags/media_tags.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from django import template
+
+from ..blocks import MediaBlock
+
+
+register = template.Library()
+
+
+@register.inclusion_tag("media/blocks/media-block--audio.html")
+def audio(media):
```
Comment (danbentley) [inline etna/media/templatetags/media_tags.py:10]: Could you add a comment to these tags?

File: etna/media/blocks.py [hunk 4]
```diff
@@ -3,34 +3,23 @@
...
-
-        # Render using the appropriate template.
-        if value.type == "audio":
-            return render_to_string("media/blocks/media-block--audio.html", context)
-        elif value.type == "video":
-            return render_to_string("media/blocks/media-block--video.html", context)
-        else:
-            return ""
+class MediaBlock(blocks.StructBlock):
+    heading = blocks.CharBlock(max_length=100, required=True)
```
Comment (danbentley) [inline etna/media/blocks.py:16]: As discussed, would you mind removing the heading field?

**Review by danbentley — CHANGES_REQUESTED** [review]

File: etna/media/blocks.py (hunk 1 above)
Comment (Rijidij) [reply etna/media/blocks.py:16]: Done

File: etna/media/templatetags/media_tags.py (hunk 3 above)
Comment (Rijidij) [reply etna/media/templatetags/media_tags.py:10]: Done

File: etna/media/blocks.py (hunk 2 above)
Comment (Rijidij) [reply etna/media/blocks.py]: Done

File: etna/media/blocks.py (hunk 4 above)
Comment (Rijidij) [reply etna/media/blocks.py]: Done

**Comment by danbentley** [comment]
Looks great! Thanks for this!

**Review by danbentley — APPROVED** [review]

**Pull request merged**